src/main/Duke.java

```diff
@@ -0,0 +1,43 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Duke {
+    private Ui ui;
+    private Storage storage;
+    private Parser parser;
+    private TaskList tasks;
+
+    protected final static String DATA_FILE_PATH = "data/dukeData.txt";
+
+    public Duke(String filePath) {
+        ui = new Ui();
+
+        ui.printLogo();
+        System.out.println("Initializing...");
+
+        storage = new Storage(filePath);
+        parser = new Parser();
+
+        try {
+            ArrayList<Task> existingTasks = storage.loadExistingData();
+            tasks = new TaskList(existingTasks);
+            ui.printWelcome(existingTasks);
+        } catch (Exception e) {
```

Comment:
In the Duke.java file, line 25, instead of catching Exception in general, you should catch specific Exceptions, so that you can identify the specific error.